test/extensions/quic_listeners/quiche/active_quic_listener_test.cc

```diff
@@ -231,7 +251,9 @@ class ActiveQuicListenerTest : public QuicMultiVersionTest {
   }
 
   void TearDown() override {
-    quic_listener_->onListenerShutdown();
+    if (quic_listener_ != nullptr) {
```

Comment:
This (as you noted) is to make it not crash with the `quic_listener_.reset()` below.

The `quic_listener_.reset()` was needed so that an assertion doesn't fail for double-registering the same listener-worker: `[source/common/network/udp_listener_impl.cc:137] assert failure: workers_.at(listener.workerIndex()) == nullptr.`

Reply:
I experimented with that, but caused a different failure because it triggers the CBPF socket option, which fails (at least on my dev machine). I played with a few other ideas, and they were all more of a mess than this, so I think we should keep this as-is.